### ISSUE TITLE
[bug] Version 1.30.2 broken copy_deps at least on Windows when ignore_case is False

### ISSUE THREAD

**Issue by cszabo2** [conan-io/conan#7990]
I had a set of working conan packages with dependencies (p1 depends on p2).
When deployed, p1 also deploys files from its dependency p2 and uses the copy_deps() method for that.
I had ignore_case = False set for all platforms and everything worked fine with conan 1.21.1 64bit but stopped working in conan 1.30.2 32 bit. The call succeeds but no files are copied.
Deploying p2 directly works, thus thre is not issue with the case of the files as the same params are used in both cases. I believe that there is a problem in computing the package path for dependencies. I believe that the path is correct except for the case.
The src parameter is empty. The pattern parameter has "include/*", which is in the correct case.
If I change "ignore_case" to True, things work.
I don't know how to debug this any further.

**Comment by memsharded** [comment]
Hi @cszabo2 

There was a bug with the default value of the ``ignore_case``, and it was fixed in Conan 1.30. The new default is ``ignore_case=True``, which was the behavior that was actually being applied, despite it had a ``ignore_case=False``. It is possible that something in your code has been affected by this.
I have tried to reproduce your issue with the information you provided, but no success so far. So we would need:

- The OS you are running this.
- The actual package names. The filenames, including casing that are packaged inside the "p2" package.
- The ``deploy()`` method of "p1"

Please attach this info and we will have another look

**Comment by cszabo2** [comment]
The exact OS is Windows 10 Enterprise. Package names are all lower case. The actual names are lua_bin and lua_cmn, version@user/channel='5.03.005.00009@unicore/official'
The actual full path to the conan cache is: C:\Users\Carol_Szabo\.conan\data
HOMEDRIVE=C:
HOMEPATH=\Users\Carol_Szabo

**Comment by memsharded** [comment]
Could you please copy your ``deploy()`` method (or at least some parts of it)?
Also, how do the files inside the dependency package, the ones to be copied, look like? Are they inside folders? Lowercase? 
Those things will also greatly help, thanks!

**Comment by memsharded** [comment]
The PR with the changes that might be causing this is: https://github.com/conan-io/conan/pull/7704

**Comment by memsharded** [comment]
Hi @cszabo2 

I am investigating this: Is it possible that you have some ``excludes`` argument in your ``self.copy``? Could you please share more details about your ``package()`` method? Thanks.

**Comment by memsharded** [comment]
I have submitted a PR to fix some other bug related to ignore_case (when using excludes): https://github.com/conan-io/conan/pull/8009

Your issue, I cannot reproduce it yet, it would be nice to have more information.

**Comment by cszabo2** [comment]
The package method is not very useful to you because all it does, it is that it calls jfrog CLI to download the content of the package from an Artifactory repository. Our packaging strategy is that for each release we upload the artifacts to an Artifactory repository, then we repackage these artifacts in various ways for various purposes, thus the package method would not work outside my company. We use conan for integration of internally developed components.

No there are no excludes. The copy_deps is called with these arguments:
``("include/*", "../alien/lua", "", root_package="lua_cmn", folder=False, ignore_case=False, excludes = None, keep_path=True)``
The "include" directory in the "lua_cmn/5.03.005.00009@unicore/official package (which is the dependency I had problems with) contains 4 header files in all lower case.

This is pretty much all I can tell you. I do not have a reduced case. I just switched between version 1.30.2 32 bit and 1.21.1 64 bit and the change in conan version caused the "include" directory to be created in the workarea but no content copied to it (it may be that the creation of the directory was a side effect of putting a version.txt file in it which the deploy() function does separately from copy_deps()) setting ignore_case to True, restores the functionality in 1.30.2.
It is not that I do not want to help, but the deploy mechanism in the packages I create is really complex and involves a helper package and calling the deploy function on a manufactured copy of the dependency conanfile module. So it would just cloud the issue.

My initial hunch is that there is a problem with computing the path of the dependency package directory. In my case this is:
C:\Users\Carol_Szabo\.conan\data\lua_cmn\5.03.005.00009\unicore\official\package\b8c6dff7bc01311f6c60fd8d2e70c99982e8d6df

If has a few uppercase letters in it. If these are in lowercase in the computed path, this may be a problem. I have no insight into how conan computes the package path, I checked the HOME* environment vars and they seem OK.

**Comment by cszabo2** [comment]
After some further investigation (I ran into the same issue with another package of ours, it looks like the ignore_case=False is probably just part of the problem. I managed to get things to work by simply moving things around.
Rather than using pattern="include/*", dst="../alien/lua", and src unspecified, I replaced the parameters like this:
pattern="*", "dst="../alien/lua/include", src="include", all other parameters being the same and things started working.
So the problem is in the pattern matching, I would think. I am not sure how the ingore_case=False plays into it.
